point3d[] vec3d
